app/utils/service_client/middleware/retry.rb

```diff
@@ -41,8 +41,10 @@ def build_retry_queue(ctx)
         ctx.fetch(:enter_queue).dup + [self]
       end
 
+      # Retries if all attempts are not used and
+      # status is 5xx
       def leave_needs_retry?(ctx)
-        !max_attempts?(ctx) && !ctx.fetch(:res).fetch(:success)
+        !max_attempts?(ctx) && (500..599).include?(ctx.fetch(:res).fetch(:status))
```

Comment:
Will this result in a sequential comparison against all numbers from 500-> one at a time? >= 500 and < 600 would be faster and simpler.